### PR TITLE
refactor: tutorial eyebrowSlot

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksrefactor-tutorial-eyebrow-hashicorp.vercel.app/swingset/components/tutorialcard) 🔎
- [Asana task](https://app.asana.com/0/1202702246158644/1202822591039668) 🎟️

## 🗒️ What

This PR refactors the `TutorialCard` to render an `eyebrowSlot`, if provided. The default will remain as the tutorial read time. 

## 🤷 Why

This decouples the bookmark logic (and in the future progress) from the tutorial card presentation, making the base component more flexible. 

## 🧪 Testing

- Visit [the swingset page](https://dev-portal-git-ksrefactor-tutorial-eyebrow-hashicorp.vercel.app/swingset/components/tutorialcard)
- scroll down to check out the example with the bookmark slot. 
- also [visit a collection](https://dev-portal-git-ksrefactor-tutorial-eyebrow-hashicorp.vercel.app/terraform/tutorials/azure-get-started) view page to see the eyebrow slot in use with the connected bookmark button. 

## 💭 Anything else?

~~This will undo the change where all tutorial cards automatically rendered the bookmark icon. However, we can adjust that once we are doing the data integration to fetch the actual bookmark state.~~ Ended up updating all instances.

### CODE REVIEW

File: src/components/tutorial-card/helpers/with-bookmark.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import { AUTH_ENABLED } from 'hooks/use-authentication'
+import { TutorialCardBookmarkButton } from 'components/bookmark-button'
+import TutorialCard from '..'
+import { TutorialCardProps } from '..'
+
+interface TutorialCardWithBookmarkProps extends TutorialCardProps {
+	isBookmarked: boolean
+}
+
+export function TutorialCardWithBookmark(props: TutorialCardWithBookmarkProps) {
```
Comment (kendallstrautman) [inline src/components/tutorial-card/helpers/with-bookmark.tsx:10]: This is a tutorial card with the eyebrowSlot composed with the connected bookmark. Eventually we will have many different types of eyebrow components to render in this slot, once we implement progress.

File: src/views/collection-view/components/collection-tutorial-list/index.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ function CollectionTutorialList({
 					const { id, ...cardProps } = tutorial
 					return (
 						<li key={tutorial.id}>
-							<TutorialCard {...cardProps} />
+							<TutorialCardWithBookmark isBookmarked={false} {...cardProps} />
```
Comment (kendallstrautman) [inline src/views/collection-view/components/collection-tutorial-list/index.tsx:18]: An example of this being used on the collection view.

File: src/components/tutorial-card/types.ts [hunk 3]
```diff
@@ -36,7 +37,7 @@ export interface TutorialCardProps {
 	 */
 	productsUsed: ProductOption[]
 
-	isBookmarked?: boolean
+	eyebrowSlot?: ReactNode // TODO figure out best type
```
Comment (kendallstrautman) [inline src/components/tutorial-card/types.ts:40]:
```suggestion
	eyebrowSlot?: ReactNode 
```

**Review by ashleemboyer — APPROVED** [review]

File: src/views/collection-view/components/collection-tutorial-list/index.tsx (hunk 2 above)
Comment (kendallstrautman) [reply src/views/collection-view/components/collection-tutorial-list/index.tsx:18]: TODO: update for all the places

File: src/components/tutorial-card/helpers/with-bookmark.tsx (hunk 1 above)
Comment (kendallstrautman) [inline src/components/tutorial-card/helpers/with-bookmark.tsx:10]:
```suggestion
export function ConnectedTutorialCard(props: TutorialCardWithBookmarkProps) {
```

**Pull request merged**